include/PassType.h

```diff
@@ -0,0 +1,51 @@
+#ifndef PASS_H
+#define PASS_H
+
+#include "RunAnalysis.h"
+#include "Utils.h"
+#include "PointerAnalysis.h"
+#include "CFG.h"
+
+#include <set>
+#include <fstream>
+
+typedef std::map<std::string, std::string> AliasMap;
+
+struct MethodsSetHolder {
+    std::set<std::string> methodsSet; 
+    bool setInitialized; 
+};
+
+typedef std::map<std::string, std::map<std::string, MethodsSetHolder>> MappedMethods;
+
+class PassType {
+private: 
+    AliasMap aliasedVars; 
+    std::string testName; 
+    CFG* cfg; 
+    MappedMethods expectedResult; 
+    
+    virtual void analyzeCFG(CFG* cfg, MappedMethods& PreMappedMethods, MappedMethods& PostMappedMethods, std::string priorBranch) = 0; 
+    virtual void transfer(Instruction* I,  SetVector<Instruction *> workSet, std::map<std::string, MethodsSetHolder>& inputMethodsSet) = 0;
+public:    
+    void setFunctions(std::map<std::string, bool> safeFunctions, std::map<std::string, bool> unsafeFunctions, std::map<std::string, bool> reallocFunctions, std::map<std::string, std::string> memoryFunctions); 
```

Comment:
Why do these need to be maps from `string->bool`? I would think just keeping the set of names (as strings) would suffice, without the need for the bool. If the bool is really needed, what it represents needs to be explained in a comment.

Reply:
I opted for using maps since I'd imagine once we further build the code, there would be a lot of safe, unsafe, realloc, & memory functions to check for so a string->bool map would help with making lookup times faster

Reply:
> I opted for using maps since I'd imagine once we further build the code, there would be a lot of safe, unsafe, realloc, & memory functions to check for so a string->bool map would help with making lookup times faster

The potential size of these maps shouldn't have an impact here: I'm asking about the data representation.

In particular, I don't understand the purpose of the `bool`. What does it represent? That is, when is it true and when is it false? It seems to me that a `std::set<std::string>` should be sufficient here: the presence of a function in the set means that it is a function of the named type; not being in the set means that it is not.